crates/provider-wit-bindgen/src/lib.rs

```diff
@@ -1749,3 +1812,174 @@ fn convert_wit_type(t: &wit_parser::Type, cfg: &ProviderBindgenConfig) -> anyhow
         }
     }
 }
+
+/// Attempt to extract key and value types from a tree of tokens that is a witified map
+///
+/// For example, the following Rust type, submitted as a list of tokens would be parsed successfully:
+///
+/// ```rust,ignore
+/// Vec<(String, String)>
+/// ```
+fn extract_witified_map_types(input: &[TokenTree]) -> Option<(TokenStream, TokenStream)> {
```

Comment:
I'm leaning toward thinking `Result` would be more clear than `Option` here. Thoughts?

I'm trying to determine if we only call this where we expect a map type to be, and therefore if we don't find one, it's an "error", vs if we optimistically call this and are okay if the following tokens don't match

Reply:
Yeah it's more that I optimistically call this if the tokens don't match I want to use the existing code/fallback, so it's not quite an "error". This is written to work at places were there *could* be a map type:

- in exported WIT functions (as the first argument or otherwise)
- in types of members of WIT-defined structs

Most of the time we'll have a clue (e.x. `process_fn_arg`) with the given naming of the argument, but that's not always the case (ex. processing structs from bindgen, someone could h).

As far as the type goes, it could be a `Result`, but an `Option` in the case of extraction failure fit a little better to me -- I'm open to changing it

Reply:
Your reasoning makes sense to me 👍 Thanks for the explanation